web/src/main/webapp/WEB-INF/classes/setup/sql/migrate/v3110/migrate-default.sql

```diff
@@ -26,3 +28,13 @@ INSERT INTO StatusValuesDes  (iddes, langid, label) VALUES (63,'spa','Record res
 INSERT INTO StatusValuesDes  (iddes, langid, label) VALUES (63,'tur','Record restored.');
 INSERT INTO StatusValuesDes  (iddes, langid, label) VALUES (63,'vie','Record restored.');
 
+DELETE FROM Settings WHERE name = 'system/server/securePort';
+
+INSERT INTO Settings (name, value, datatype, position, internal) VALUES ('system/security/passwordEnforcement/minLength', '6', 1, 12000, 'n');
```

Comment:
We have an issue here, because it will run it once here and then again in https://github.com/geonetwork/core-geonetwork/blob/main/web/src/main/webapp/WEB-INF/classes/setup/sql/migrate/v403/migrate-default.sql#L4 and the second one will fail no ?

Reply:
@fxprunayre - I believe you are correct but I'm not sure what should be done.
If I remove them and someone upgrades from 4.0.0, 4.0.1 or 4.0.2 then they will not get the changes. If we keep them there then it will fail as you indicated.

I believe the only way to fix it is to only issue the insert if the data does not already exists. And to do that check, it would need to be coded in Java and that seems like overkill.

Any ideas on how to proceed?

Reply:
Same issue happens with language related values - if in your installation, you remove a language, then migration will fails most of the time. 

Not sure, we can probably workaround with something like 
```sql
INSERT INTO Settings (name, value, datatype, position, internal)
  SELECT 'system/security/passwordEnforcement/minLength', '6', 1, 12000, 'n'
  WHERE (SELECT count(*) FROM Settings WHERE name = 'system/security/passwordEnforcement/minLength') = 0;
```
which should be quite ok in all db types ?

Indeed all in Java seems overkill

A conditional check that we run before ?

```sql
-- IF SELECT count(*) FROM Settings WHERE name = 'system/security/passwordEnforcement/minLength'
INSERT INTO Settings (name, value, datatype, position, internal) VALUES ('system/security/passwordEnforcement/minLength', '6', 1, 12000, 'n');
```

I think @josegar74 talked long time ago about some tools to help on this - but I don't remember the name. Maybe quite a job to setup.

Reply:
@fxprunayre 
The insert statement that you identified will not work on Oracle because it would require "from dual"

I believe the following would be more compatible with other databases - But I did not test them all...

```
INSERT INTO Settings (name, value, datatype, position, internal) 
SELECT distinct 'system/security/passwordEnforcement/minLength', '6', 1, 12000, 'n' from settings 
WHERE NOT EXISTS (SELECT name FROM Settings WHERE name = 'system/security/passwordEnforcement/minLength');
```

I have updated the PR to incorporate these changes.